playkit/src/main/java/com/kaltura/playkit/LocalAssetsManager.java

```diff
@@ -211,22 +203,7 @@ public void unregisterAsset(@NonNull final String localAssetPath,
 
         final DrmAdapter drmAdapter = DrmAdapter.getDrmAdapter(scheme, context, localDataStore);
 
-        doInBackground(new Runnable() {
-            @Override
-            public void run() {
-                drmAdapter.unregisterAsset(localAssetPath, assetId, new AssetRemovalListener() {
-                    @Override
-                    public void onRemoved(final String localAssetPath) {
-                        mainHandler.post(new Runnable() {
-                            @Override
-                            public void run() {
-                                removeAsset(localAssetPath, assetId, listener);
-                            }
-                        });
-                    }
-                });
-            }
-        });
+        doInBackground(() -> drmAdapter.unregisterAsset(localAssetPath, assetId, localAssetPath1 -> mainHandler.post(() -> removeAsset(localAssetPath1, assetId, listener))));
```

Comment:
Newlines to make it clear we're running something in the background and then in the main thread.